File: src/main/java/org/example/logica/StrReverse.java
package org.example.logica;

public class StrReverse {

    public static void main(String[] args) {
        String str = "RAFAEL";
        var reversedStr = reverseString(str);
        System.out.println(reversedStr);

    }

    private static Object reverseString(String str) {
        StringBuilder reversed = new StringBuilder();
        for (int i = str.length() - 1; i >= 0; i--) {
            reversed.append(str.charAt(i));
        }
        return reversed.toString();
    }

}

File: src/main/java/org/example/just_kotlin/StrReverse.kt
package org.example.just_kotlin

fun main() {
    val str = "rafael";
    val reverseStr = StrReverse().reverseString(str);
    println(reverseStr);
}

class StrReverse {
    fun reverseString(str: String): String {
        var reverseStr = ""
        for (i in str.length -1 downTo 0) {
            reverseStr += str[i]
        }
        return reverseStr
    }
}